iwyu_test_util.py

```diff
@@ -73,12 +74,69 @@ class Features:
   # Parses the output of 'include-what-you-use -print-targets'.
   SUPPORTED_TARGET_RE = re.compile(r'^\s+([a-z0-9_-]+)\s* - .*$')
 
+  # Parses the output of 'include-what-you-use -dM -E'.
+  PREPROCESSOR_MACRO_RE = re.compile(r'^#define ([A-z_][A-z0-9_]*)')
+
   def __init__(self):
     # Maps all known features names to prerequisite predicates.
     self.known = {
-      'has-target': Features.HasTarget
+      'has-target': Features.HasTarget,
+      'has-c-predefined-macro': Features.HasCPredefinedMacro,
+      'has-cxx-predefined-macro': Features.HasCXXPredefinedMacro,
```

Comment:
Let's not add these `has-...predefined-macro` features in this PR, and focus on one feature at a time.